app/middleware/auth.py

```diff
@@ -0,0 +1,29 @@
+from fastapi import Request, HTTPException
+import httpx
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+PO_AUTH_ROUTE = os.getenv("PO_AUTH_ROUTE")
+if PO_AUTH_ROUTE is None:
+    raise RuntimeError("PO_AUTH_ROUTE environment variable is not set.")
+
+
+async def auth_dependency(request: Request):
```

Comment:
For caching reason, you will have to split this function with a function that takes session_cookie as parameter and do the request part.